src/CoreWCF.Kafka.Client/src/CoreWCF/ServiceModel/Channels/KafkaOutputChannel.cs

```diff
@@ -127,7 +127,7 @@ private async Task SendAsync(Message message, TimeSpan timeout)
                 kafkaMessage.Value = new Span<byte>(messageBuffer.Array, messageBuffer.Offset, messageBuffer.Count).ToArray();
                 await _producer.ProduceAsync(_topic, kafkaMessage, cts.Token);
             }
-            catch (ProduceException<Null, byte[]> produceException)
+            catch (ProduceException<byte[], byte[]> produceException)
             {
```

Comment:
Is it no longer possible for `ProduceException<Null, byte[]>` to get thrown? Is there a non-generic base type to ProduceException that might better fit here?

Reply:
I see the base type for `ProduceException<TKey, TValue>` is `KafkaException`. It looks like the only property we use from `ProduceException<TKey, TValue>` is `Error`, which lives on the base exception `KafkaException`. Would it make sense to switch the catch and the helper method to using KafkaException so we don't care what the generic shape is?

Reply:
KafkaException is now caught